hedera-node/hedera-app/src/main/java/com/hedera/node/app/config/internal/ConfigurationAdaptor.java

```diff
@@ -0,0 +1,359 @@
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.config.internal;
+
+import static com.hedera.node.app.spi.config.PropertyNames.LEDGER_FUNDING_ACCOUNT;
+
+import com.hedera.hapi.node.base.AccountID;
+import com.hedera.node.app.service.mono.config.HederaNumbers;
+import com.hedera.node.app.service.mono.context.properties.PropertySource;
+import com.hedera.node.app.spi.config.GlobalConfig;
+import com.hedera.node.app.spi.config.GlobalDynamicConfig;
+import com.hedera.node.app.spi.config.NodeConfig;
+import com.hedera.node.app.spi.config.Profile;
+import com.hedera.node.app.spi.config.PropertyNames;
+import com.swirlds.config.api.Configuration;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Collection;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+
+/**
+ * Adaptor for the configuration functionality. This class will be removed in future once the full services layer is
+ * refactored to use the "real" config api from the platform. In that case the {@link Configuration} implementation from
+ * the platform will be used.
+ *
+ * <p>This implementation is backed by a {@link PropertySource} instance and all calls will be
+ * forwarded to that instance.
+ */
+@Deprecated(forRemoval = true)
+public class ConfigurationAdaptor implements Configuration {
+
+    private final PropertySource propertySource;
+
+    private final HederaNumbers hederaNums;
+
+    private final NodeConfig nodeConfig;
+
+    private final GlobalConfig globalConfig;
+
+    private final GlobalDynamicConfig globalDynamicConfig;
+
+    public ConfigurationAdaptor(@NonNull final PropertySource propertySource) {
+        this.propertySource = Objects.requireNonNull(propertySource, "propertySource");
+        this.hederaNums = new HederaNumbers(propertySource);
+        nodeConfig = createNodeConfig();
+        globalConfig = createGlobalConfig();
+        globalDynamicConfig = createGlobalDynamicConfig();
+    }
+
+    @Override
+    public Stream<String> getPropertyNames() {
+        return propertySource.allPropertyNames().stream();
+    }
+
+    @Override
+    public boolean exists(final String name) {
+        return propertySource.containsProperty(name);
+    }
+
+    @Override
+    public String getValue(final String name) throws NoSuchElementException {
+        if (exists(name)) {
+            return propertySource.getRawValue(name);
+        }
+        throw new NoSuchElementException(exceptionMessagePropertyDoesNotExist(name));
+    }
+
+    @Override
+    public String getValue(final String name, final String defaultValue) {
+        if (exists(name)) {
+            return getValue(name);
+        }
+        return defaultValue;
+    }
+
+    @Override
+    public <T> T getValue(final String name, final Class<T> type)
+            throws NoSuchElementException, IllegalArgumentException {
```

Comment:
Curious what the purpose is for putting a `RuntimeException` in a `throws` clause.  Inform the reader, or?

Reply:
yes, it is to inform the reader and have it in the javadoc

Reply:
(Is this a general practice in this codebase that runtime exceptions are documented in the API via `throws`?  Possibly I just haven't run across it yet.)